test/GoodGhostingPolygon.test.js

```diff
@@ -904,6 +904,212 @@ contract("GoodGhostingPolygon", (accounts) => {
+        beforeEach(async () => {
+            pap = await LendingPoolAddressesProviderMock.new("TOKEN_NAME", "TOKEN_SYMBOL", { from: admin });
+            aToken = await IERC20.at(await pap.getLendingPool.call());
+            await pap.setUnderlyingAssetAddress(incentiveController.address);
+            await incentiveController.mint(player1, toWad(1000), { from: admin });
+            await incentiveController.mint(player2, toWad(1000), { from: admin });
+            contract = await GoodGhostingPolygon.new(
+                incentiveController.address,
+                pap.address,
+                segmentCount,
+                segmentLength,
+                segmentPayment,
+                fee,
+                0,
+                pap.address,
+                "115792089237316195423570985008687907853269984665640564039457584007913129639935", // equals to 2**256-1
+                ZERO_ADDRESS,
+                incentiveController.address,
+                incentiveController.address,
+                { from: admin },
+            );
+        });
+
+        it("ensures the reward params are set correctly after redeem", async () => {
+            const expectedBalance = web3.utils.toBN(segmentPayment * segmentCount * 2);
+            await incentiveController.approve(contract.address, segmentPayment, { from: player1 });
+            await incentiveController.approve(contract.address, segmentPayment, { from: player2 });
+
+            await contract.joinGame( { from: player2 });
+            await contract.joinGame( { from: player1 });
+            for (let index = 1; index < segmentCount; index++) {
+                await timeMachine.advanceTime(weekInSecs);
+                await incentiveController.approve(contract.address, segmentPayment, { from: player1 });
+                await incentiveController.approve(contract.address, segmentPayment, { from: player2 });
+
+                await contract.makeDeposit({ from: player1 });
+                await contract.makeDeposit({ from: player2 });
+
+            }
+            // above, it accounted for 1st deposit window, and then the loop runs till segmentCount - 1.
+            // now, we move 2 more segments (segmentCount-1 and segmentCount) to complete the game.
+            await timeMachine.advanceTime(weekInSecs);
+            await timeMachine.advanceTime(weekInSecs);
+            let contractMaticBalanceBeforeRedeem = await incentiveController.balanceOf(contract.address);
+            await contract.redeemFromExternalPool({ from: player2 });
+            let contractMaticBalanceAfterRedeem = await incentiveController.balanceOf(contract.address);
+            const rewardBalance = await contract.rewardsPerPlayer.call();
+            assert(rewardBalance.eq(new BN(0)))
+            assert(contractMaticBalanceAfterRedeem.gt(contractMaticBalanceBeforeRedeem));
+            const contractsDaiBalance = await incentiveController.balanceOf(contract.address);
```

Comment:
after redeem, shouldn't we get the balance from the GG contract instead of the "incentiveController"?
This applies to all the new tests

Reply:
@fodisi the incentive controller address in unit tests is also the mock matic address since on withdrawing the mock incentive controller mints the matic tokens too hence